src/openvic-simulation/pop/Pop.hpp

```diff
@@ -90,23 +102,37 @@ namespace OpenVic {
 		fixed_point_t PROPERTY(income);
 		fixed_point_t PROPERTY(expenses); //positive value means POP paid for goods. This is displayed * -1 in UI.
 		fixed_point_t PROPERTY(savings);
-		fixed_point_t PROPERTY(life_needs_fulfilled);
-		fixed_point_t PROPERTY(everyday_needs_fulfilled);
-		fixed_point_t PROPERTY(luxury_needs_fulfilled);
+
+		#define NEED_MEMBERS(need_category)\
+			fixed_point_t need_category##_needs_acquired_quantity, need_category##_needs_desired_quantity;\
+			constexpr fixed_point_t get_##need_category##_needs_fulfilled() const {\
+				if (need_category##_needs_desired_quantity == fixed_point_t::_0()) {\
+					return fixed_point_t::_1();\
+				}\
+				return need_category##_needs_acquired_quantity / need_category##_needs_desired_quantity;\
+			}\
+			GoodDefinition::good_definition_map_t need_category##_needs {};\
+			ordered_map<GoodDefinition const*, bool> PROPERTY(need_category##_needs_fulfilled_goods);
```

Comment:
```suggestion
			fixed_point_t need_category##_needs_acquired_quantity, need_category##_needs_desired_quantity; \
		public: \
			constexpr fixed_point_t get_##need_category##_needs_fulfilled() const { \
				if (need_category##_needs_desired_quantity == fixed_point_t::_0()) { \
					return fixed_point_t::_1(); \
				} \
				return need_category##_needs_acquired_quantity / need_category##_needs_desired_quantity; \
			} \
		private: \
			GoodDefinition::good_definition_map_t need_category##_needs {}; \
			ordered_map<GoodDefinition const*, bool> PROPERTY(need_category##_needs_fulfilled_goods);
```
`get_##need_category##_needs_fulfilled()` needs to be public so that the population menu UI code can access it (also better to have spaces before `\`s at the end of lines)

Reply:
`get_##need_category##_needs_fulfilled()` still needs to be public, add `public:` above it and `private` below it in the macro